compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/symbols/Symbol.java

```diff
@@ -66,4 +66,12 @@
      * @return The location of the symbol if applicable
      */
     Optional<Location> getLocation();
+
+    /**
+     * Check whether the given name equals to the name of the symbol.
+     *
+     * @param name The name that needs to be compared
+     * @return true if the name is similar to the symbol's name and false otherwise
+     */
+    boolean nameEquals(Optional<String> name);
```

Comment:
Shall we not use `Optional` in the param here? Anyone who wants to compare a name would have to first wrap it in `Optional`. Plus use of `Optional` in params is not the intended use for it.

Reply:
Also, let's elaborate a bit more in the API docs on what exactly this method does: comparing the symbol's name against the given name, according to the semantics defined in the spec regarding identifiers.